starfish/spots/_detector/pixel_spot_detector.py

```diff
@@ -37,7 +40,10 @@ def __init__(
             an ImageStack (default = 0)
 
         """
-        self.codebook = codebook
+        if isinstance(codebook, str):
```

Comment:
We have churned on this before, and the agreement we came to is that this complexity should be banished to CLI-specific methods (`run`), since users are unlikely to need to load the `codebook` from a string. Is there a reason this isn't possible here?